### ISSUE TITLE
Cannot add an auto-export

### ISSUE THREAD

**Issue by awpzs** [retorquere/zotero-better-bibtex#490]
Report ID: 5XWNRT9N

After exporting a collection with *keep updated* checked, the export process runs well and the exported collection appears properly in *Preferences - Better BibTex - Automatic export*. However, after restarting Zotero, the exported collection just disappeared from *Automatic export*.

What could go wrong here? Thanks in advance.

**Comment by retorquere** [comment]
That means the DB isn't saved. Could you try https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.51-circle-2382.xpi ?

**Comment by awpzs** [comment]
Thank you, but the problem still exists.

**Comment by retorquere** [comment]
Can you

* go into the BBT prefs, section Advanced, enable debugging
* add a new auto-export
* from the gear menu, select "send better bibtex error"
* post the error ID (red) here

**Comment by awpzs** [comment]
Yeah sure, please check Report ID: S2TVTXDF .

**Comment by retorquere** [comment]
Could you re-install the [latest release](https://github.com/retorquere/zotero-better-bibtex/releases/tag/1.6.51) and do that again?

**Comment by awpzs** [comment]
Yes, the new Report ID is XWMV2RVF .

**Comment by retorquere** [comment]
I'm sorry to have to subject to this, but I need a full log of what's happening: can you get a "Real-time Debug Output" as per https://www.zotero.org/support/debug_output (which is a pain on Windows, I'm so sorry about that) and drop the resulting log into https://dbinbox.com/retorquere ? Your DB isn't saved on shutdown for some reason, and that reason won't be in the usual log.

**Comment by awpzs** [comment]
Yeah I guess there may be something wrong with Zotero itself, but reinstalling Zotero and BBT does not fix this problem either. I've already dropped the debug log into your dbinbox and I hope that will be helpful.

**Comment by awpzs** [comment]
Oops sorry the log I sent was not a real-time output, I will send you the correct one immediately.

**Comment by retorquere** [comment]
Super -- please make sure you capture the log after you've closed Zotero, as I'm looking for the shutdown event.

**Comment by awpzs** [comment]
I'm afraid that it is very difficult to capture the real-time log which you needed - the Windows console only allows me to copy part of its output (the older output just disappears and cannot be copied) and it rapidly shuts down after closing Zotero. I will let you know if I could figure out anyway to capture the log.

**Comment by retorquere** [comment]
Does it work if you add ` > c:\zotero.log` to the end of that command?

**Comment by awpzs** [comment]
No, it generates a `zotero.log` but does not write anything into it.

**Comment by retorquere** [comment]
Dows adding ` 2>&1` after that change that?

**Comment by awpzs** [comment]
No it doesn't :(

**Comment by retorquere** [comment]
Windows really makes this more of a PITA than it needs to be. 

There *is* a way, but it's even more unpleasant than what you already tried. You could try installing cygwin and script according to these steps (http://superuser.com/questions/325540/is-there-a-windows-equivalent-of-the-script-command-in-linux) and then fire up "bash" , within that "script", and within *that* the former command without the `>` redirects. After Firefox is done, type "exit", and you'll be left with a full log in a file named "typescript". 

I've searched for a less invasive solution, but I have not been able to find one. Also still looking at reasons why the db doesn't save, but my current hypothesis is that something errors out before it gets to the code that does the save, and it's really hard to guess what.

**Comment by retorquere** [comment]
Ah wait - if you run "cmd", and then run Firefox from there, the command she'll will probably stay open. If we get lucky, the error will be in the part that's still on-screen.

**Comment by awpzs** [comment]
Thank you very much for your suggestions. I am using Zotero Standalone and I have already tried running Zotero using cmd, but that would start a new window for ZoteroDebug and that window would close quickly after shutting down Zotero. Anyway, I'll try cygwin and see if it works.

**Comment by retorquere** [comment]
Crap, if that's the case it's unlikely that cygwin will help. I'll look for
another solution.

**Comment by perspexi** [comment]
I have the same problem. I made an WME, it's rather large (250MB). Hope it helps. https://goo.gl/lBmyRJ

**Comment by retorquere** [comment]
I don't have a Windows PC.

**Comment by retorquere** [comment]
Did you use `-jsconsole` or `-console`? If you previously used `-jsconsole`, could you try

```
"C:\Program Files (x86)\Mozilla Firefox\firefox.exe" -console -ZoteroDebug > c:\zotero.log
```

(remove the `(x86)` as necessary).

**Comment by retorquere** [comment]
Could you give https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.52-circle-2396.xpi a try, with debugging enabled, and after adding an auto-export, letting it sit for 10 seconds, then submit a new error report?

**Comment by awpzs** [comment]
Yes, please see Report ID: UNZ5SDFF .

**Comment by retorquere** [comment]
This is so strange, I can see the autoexport being saved after it's added. Could you try again with https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.52-circle-2397.xpi and submit a new log?

**Comment by awpzs** [comment]
Yes. Please check SVUMKW86 .

**Comment by retorquere** [comment]
I think I have it! FFS, if this really is it (and it must be), you've caught me at my stupidest. Thanks so much for reporting this!

Try https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.52-circle-2400.xpi

**Comment by awpzs** [comment]
It works! Thank you so much for your help! :+1:

**Comment by retorquere** [comment]
Errrr.... I should be thanking you. This could have been really, really bad. What was confusing is that BBT really *did* save these settings. It just didn't *load* them on startup. I'm cutting a new release right now.

**Comment by retorquere** [comment]
.53 is out.

**Comment by perspexi** [comment]
Thank you!!

**Comment by ETaSky** [comment]
Hello retorquere,

It appears that this issue happens again in the latest version of BetterBibTex (.68). The auto export disappeared after zotero restarted. I am using Zotero standalone v 4.0.29.10 on a windows machine. I tried to install previous versions of Better BibTex (.53, .57, .66, .67) to see if the issue disappeared. But to make it even weirder, in all the previous versions I have tested, when clicking "export collections", the BetterBibTex format was not there (so as the checkbox of "Keep Updated").
Thank you!
